Core/Relationships/RelationshipResolver.cs

```diff
@@ -56,63 +57,126 @@ public object Clone()
     }
 
     // TODO: RR currently conducts a depth-first resolution of requirements. While we do the
-    // right thing in processing all depdenencies first, then recommends, and then suggests,
+    // right thing in processing all dependencies first, then recommends, and then suggests,
     // we could find that a recommendation many layers deep prevents a recommendation in the
     // original mod's recommends list.
     //
     // If we resolved in things breadth-first order, we're less likely to encounter surprises
     // where a nth-deep recommend blocks a top-level recommend.
 
-    // TODO: Add mechanism so that clients can add mods with relationshup other than UserAdded.
-    // Currently only made to support the with_{} options.
+    // TODO: Add mechanism so that clients can add mods with relationshup other than UserAdded. 
+    // Currently only made to support the with_{} options. 
+
+
+    /// <summary>
+    /// A class used to resolve relationships between mods. Primarily used to satisfy missing dependencies and to check for conflicts on proposed installs. 
+    /// </summary>
+    /// <remarks>
+    /// All constructors start with currently installed modules, to remove <see cref="RelationshipResolver.RemoveModsFromInstalledList" />  
+    /// </remarks>
     public class RelationshipResolver
     {
         // A list of all the mods we're going to install.
         private static readonly ILog log = LogManager.GetLogger(typeof (RelationshipResolver));
         private readonly Dictionary<string, CkanModule> modlist = new Dictionary<string, CkanModule>();
-
-        private readonly List<KeyValuePair<Module, Module>> conflicts =
+        private readonly List<CkanModule> user_requested_mods = new List<CkanModule>();
+        private readonly List<KeyValuePair<Module, Module>> conflicts = 
             new List<KeyValuePair<Module, Module>>();
+        private readonly Dictionary<Module, SelectionReason> reasons = 
+            new Dictionary<Module, SelectionReason>(new Module.IdentifierEqualilty());
```

Comment:
Sweet! Today I learnt you can pass an equality comparison class to dictionaries. :)